aiohttp/client_reqrep.py

```diff
@@ -343,11 +343,8 @@ def update_headers(self, headers: Optional[LooseHeaders]) -> None:
                     self.headers.add(key, value)
 
     def update_auto_headers(self, skip_auto_headers: Iterable[str]) -> None:
-        self.skip_auto_headers = CIMultiDict(
-            (hdr, None) for hdr in sorted(skip_auto_headers)
-        )
-        used_headers = self.headers.copy()
-        used_headers.extend(self.skip_auto_headers)  # type: ignore[arg-type]
+        self.skip_auto_headers = set(skip_auto_headers)
+        used_headers = self.skip_auto_headers | self.headers.keys()
```

Comment:
Please sanity check this change.

There was a type error, but I see no reason to be using a dict here, all references to it only use an `in` test. The same name attribute in client.py is also a set().